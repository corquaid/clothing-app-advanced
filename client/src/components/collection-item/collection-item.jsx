import React from 'react';

import { connect } from 'react-redux';

import CustomButton from '../custom-button/custom-button'
import { addItem } from '../../redux/cart/cart.actions';

import './collection-item.scss';

const CollectionItem = ({ item, addItem }) => {
  const { name, price, imageUrl } = item; // desctructure values from the item object

  return(
  <div className='collection-item'>
    <div 
      className='image'
      style={{
        backgroundImage: `url(${imageUrl})`
      }}      
    />
    <div className='collection-footer'>
      <span className='name'>{ name }</span>
      <span className='price'>{ price }</span>
    </div>
  <CustomButton className='custom-button' onClick={() => addItem(item)} inverted> Add to Cart</CustomButton> 
  {/* addItem function is fired on each button click */}
  </div>
  )
};

const mapDispatchToProps = dispatch => ({ // addItem will now be accessible in the props
  addItem: item => dispatch(addItem(item))
});

export default connect(null, mapDispatchToProps)(CollectionItem);