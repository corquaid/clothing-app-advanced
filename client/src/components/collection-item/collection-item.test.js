import React from "react";
import { shallow } from "enzyme";
import { CollectionItem } from "./collection-item";

describe("CollectionItem component testing", () => {
    let wrapper;
    let mockAddItem;
    const mockUrl = 'www.photo.com';
    const mockName = 'scarf';
    const mockPrice = 50;

    beforeEach(() => {
        mockAddItem = jest.fn();

        const mockProps = {
            addItem: mockAddItem,
            item: {
                name: mockName,
                price: mockPrice,
                imageUrl: mockUrl,
            },
        };

        wrapper = shallow(<CollectionItem {...mockProps} />);
    });

    it('should render the CollectionItem component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should call addItem when button is clicked', () => {
        wrapper.find('[id="add-button"]').simulate('click')
        expect(mockAddItem).toHaveBeenCalled();
    })

    it('should render imageUrl as a prop BackgroundImage', () => {
        expect(wrapper.find('.image').prop('imageUrl')).toBe(mockUrl)
    })
    it('should render the name prop in NameContainer', () => {
        expect(wrapper.find('.name').text()).toBe(mockName)
    })
    it('should render the price prop in PriceContainer', () => {
        // dealing with integers!!!
        const price = parseInt(wrapper.find('.price').text())
        expect(price).toBe(mockPrice)
    })
});
