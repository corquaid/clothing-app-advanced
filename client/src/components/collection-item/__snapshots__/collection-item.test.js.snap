// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component testing should render the CollectionItem component 1`] = `ShallowWrapper {}`;
